internal/authorization/helpers.go

```diff
@@ -0,0 +1,19 @@
+package authorization
+
+import (
+	"context"
+)
+
+type ClientAuthorizer = authorizer
+
+func (a *ClientAuthorizer) ClientResource(clientID string) string {
+	return "client:" + clientID
+}
+
+func (a *ClientAuthorizer) CreateClient(ctx context.Context, clientID string) error {
+	user := ctx.Value(USER_CTX)
+	if user == nil {
+		return nil
+	}
+	return a.client.WriteTuple(ctx, "user:"+user.(*User).ID, CAN_VIEW, a.ClientResource(clientID))
```

Comment:
suggestion: You lack the check for the typing to `*User` being successful.
Taking a look at [this](https://github.com/canonical/identity-platform-admin-ui/blob/main/pkg/authentication/oidc.go#L64-L66), you could improve this and add the check to make sure that `user.(*User)` goes well by doing:

```suggestion
	if user, ok := ctx.Value(USER_CTX).(*User); !ok {
		return nil
	}
	
	return a.client.WriteTuple(ctx, "user:"+user.ID, CAN_VIEW, a.ClientResource(clientID))
```

This way when you use the `user` object you know 100% it is a legitimate `*User` type.